src/test/java/ru/curs/xylophone/TestReader.java

```diff
@@ -196,6 +201,53 @@ public void testSAXReader3() throws XML2SpreadSheetError {
 		reader.process();
 		assertEquals("Q{TCQ{CCQ{CC}C}TQ{CQh{CCC}}TQ{}}F", w.getLog().toString());
 	}
+
+	@Test
+	public void testParsingDescriptorWithElementInsideElementShouldFail() throws XML2SpreadSheetError {
+		descrStream = TestReader.class
+				.getResourceAsStream("testdescriptor_with_element_inside_element.xml");
+		dataStream = TestReader.class.getResourceAsStream("testdata.xml");
+
+		expectedException.expect(XML2SpreadSheetError.class);
+		expectedException.expectMessage("Tag <element> is not allowed inside <element>. Error inside element with name titlepage.");
+
+		DummyWriter w = new DummyWriter();
+		// При создании reader (при парсинге xml) падает исключение о некорректной последовательности тегов.
+		XMLDataReader.createReader(dataStream,
+				descrStream, false, w);
+	}
+
+	@Test
+	public void testParsingDescriptorWithIterationInsideIterationShouldFail() throws XML2SpreadSheetError {
+		descrStream = TestReader.class
+				.getResourceAsStream("testdescriptor_with_iteration_inside_iteration.xml");
+		dataStream = TestReader.class.getResourceAsStream("testdata.xml");
+
+		expectedException.expect(XML2SpreadSheetError.class);
+		expectedException.expectMessage("Tag <iteration> is not allowed inside <iteration>. " +
+				"Error inside element with name titlepage.");
+
+		DummyWriter w = new DummyWriter();
+		// При создании reader (при парсинге xml) падает исключение о некорректной последовательности тегов.

```

Comment:
Xylophone is opensource project. Please, use english language in comments and javadocs.